### ISSUE TITLE
Replace your Dockerfile with go Guide + Demo Video 

### ISSUE THREAD

**Issue by mircubed** [dagger/dagger#3426]
A guide that shows how to replace a Dockerfile with Go 

**To-do:** 
- [ ] Guide 
- [ ] Demo Video (don't start recording until guide content is approved)

**Comment by vikram-dagger** [comment]
Is this a duplicate of #3614 ?

**Comment by mircubed** [comment]
@vikram-dagger yes, but let's close #3614 because it was created later. I will reassign to you, but let @kpenfound know what you need! 

Once the guide is done, I will let you choose if you want to do the demo video or if you'd like Kyle to do it.

**Comment by vikram-dagger** [comment]
@kpenfound @mircubed one idea is to take a standard, popular open source app that is available as an official Docker image and replace its Dockerfile with our pipeline. One example might be memcached: https://github.com/docker-library/memcached/blob/master/alpine/Dockerfile (but it could also be wordpress, mariadb or something else). WDYT? Any other approach?

In terms of support for the above idea, I would need the code that replicates the Dockerfile operations in the Dagger Go SDK.

**Comment by mircubed** [comment]
Yeah, that sounds great! I think that it would probably be related to #3584 . I'll let @kpenfound and you figure out which would be best to use

**Comment by mircubed** [comment]
@vikram-dagger @kpenfound This is a really popular ask from the community, so we should make sure to cover this during a community call and maybe even make a blog post to go into more detail around the use case. I'll add it to the community call backlog as a topic
